.github/workflows/test.yaml

```
@@ -1,5 +1,11 @@
 name: Test and Validate
-on: [push]
+on:
+  push:
+    paths:
+      - "**.go"
+  pull_request:
+    paths:
+      - "**.go"
```

There are a few other relevant files where you'd want this to trigger, such as `.github/workflows/test.yaml`, `.golangci.yml`, `go.mod`, `go.sum`, `Makefile`, so maybe easier to [reverse](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore) the pattern?

```suggestion
on:
  push:
    paths-ignore:
      - '**.md'
      - '.github/ISSUE_TEMPLATE/**'
  pull_request:
    paths-ignore:
      - '**.md'
      - '.github/ISSUE_TEMPLATE/**'
```